crates/bevy_ecs/src/world/entity_ref.rs

```diff
@@ -85,6 +85,21 @@ impl<'w> EntityRef<'w> {
         }
     }
 
+    /// Retrieves the change ticks for the given [`ComponentId`]. This can be useful for implementing change
+    /// detection in custom runtimes.
+    ///
+    /// **You should prefer to use the typed API where possible and only
+    /// use this in cases where the actual component types are not known at
+    /// compile time.**
+    #[inline]
+    pub fn get_change_ticks_by_id(&self, component_id: ComponentId) -> Option<&'w ComponentTicks> {
```

Comment:
These impls aren't safe. `get_ticks` assumes `component_id` is valid. `get_ticks_with_type` validates this by looking up the ComponentId using the TypeId (and returning None if there is no valid mapping).

Reply:
Thanks for the feedback @cart 
What is the preferred course of action in this case?

I assume it is better to add the following validation, just double-checking before making another commit.
```rust
if !self.contains_id(component_id) {
    return None;
}
```

Alternatively, I could make the methods unsafe and delegate the safety validation to the end user.

Reply:
The other `*_by_id` APIs are also all `unsafe` and require user validation. It might be best to just mark it unsafe and document the safety invariants.

Reply:
Strong preference to keep these safe :) The extra expressiveness isn't useful here, just a footgun.